src/csync/ConfigureChecks.cmake

```diff
@@ -22,6 +22,11 @@ if (NOT LINUX)
     check_library_exists(rt nanosleep "" HAVE_LIBRT)
 
     set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} )
+
+    # Systems not using glibc require linker flag for argp
+    if(HAVE_ARGP_H)
+        set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} argp)
```

Comment:
you are adding argp library to the linker, please check for it before

Reply:
In FreeBSD if the include file exists the library also exists.

Anyway I can add that but I have very little experience with cmake. Could you just point me in the right direction to find documentation on how to perform the check?

Reply:
I prefer to be careful
so you can do it like it is done for librt
```
check_library_exists(argp "" HAVE_LIBARGP)
```